### PR TITLE
Add Versions as a return argument to Get and Put APIs

### PR DESCRIPTION
**Pull request by adityadani**
**What this PR does / why we need it**:
- All the secret providers which support secret versioning will return the version number of the secret as part of Get and Put APIs.
- For secret providers which do not support versioning, they will return a special version call "NoVersion".

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: azure/azure_kv.go [hunk 1]
```diff
@@ -120,28 +120,30 @@ func (az *azureSecrets) GetSecret(
 		secretData[secretID] = *secretResp.Value
 	}
 
-	return secretData, nil
+	// TODO: Azure does not support version numbers, but we could leverage LastUpdateTime as an indicator
```
Comment (dahuang-purestorage) [inline azure/azure_kv.go:123]: could we create a ticket to track this?

File: k8s/k8s_integration_test.go [hunk 2]
```diff
@@ -53,26 +54,26 @@ func (k *k8sSecretTest) TestPutSecret(t *testing.T) error {
 	k.passphrase = uuid.New()
 	secretData[secretId] = k.passphrase
 	// PutSecret with non-nil secretData and no namespace should fail
-	err := k.s.PutSecret(secretName, secretData, nil)
+	_, err := k.s.PutSecret(secretName, secretData, nil)
```
Comment (dahuang-purestorage) [inline k8s/k8s_integration_test.go:57]: is there a reason we are not testing the secret version for k8s?

**Review by dahuang-purestorage — APPROVED** [review]
just the comment on test. everything else LGTM

File: k8s/k8s_integration_test.go (hunk 2 above)
Comment (adityadani) [reply k8s/k8s_integration_test.go:63]: Fixed the k8s integration tests

**Pull request merged**